apps/web/src/components/ui/navigation.tsx

```diff
@@ -23,10 +24,42 @@ export function Navigation() {
   const pathname = usePathname();
   const featureFlags = useContext(FeatureFlagContext);
 
+  const clamp = (number: number, min: number, max: number) => Math.min(Math.max(number, min), max);
+
+  function useBoundedScroll(bounds: number) {
+    const { scrollY } = useScroll();
+    const scrollYBounded = useMotionValue(0);
+    const scrollYBoundedProgress = useTransform(scrollYBounded, [0, bounds], [0, 1]);
+
+    useEffect(() => {
+      return scrollY.onChange((current) => {
+        const previous = scrollY.getPrevious();
+        const diff = current - previous;
+        const newScrollYBounded = scrollYBounded.get() + diff;
+
+        scrollYBounded.set(clamp(newScrollYBounded, 0, bounds));
+      });
+    }, [bounds, scrollY, scrollYBounded]);
+
+    return { scrollYBounded, scrollYBoundedProgress };
+  }
+
+  const { scrollYBoundedProgress } = useBoundedScroll(400);
+  const scrollYBoundedProgressThrottled = useTransform(
+    scrollYBoundedProgress,
+    [0, 0.75, 1],
+    [0, 0, 1],
+  );
+
   return (
-    <header className="z-10 w-full">
+    <motion.header
+      className="fixed inset-x-0 z-20 flex w-full overflow-hidden bg-white/50 backdrop-blur-md dark:bg-black/50"
+      style={{
+        height: useTransform(scrollYBoundedProgressThrottled, [0, 1], [80, 0]),
```

Comment:
changing height of the element is not ideal for fps as it triggers a bunch of repaints, I'd `translate-y` it off the screen instead

https://github.com/bautistaaa/typehero/assets/31113245/a8631154-6d79-4258-a7d0-3d770c7036e4

Reply:
would look even nicer if it's like `scale(0.9)` and `opacity: 0` alongside `translate-y`, none of those trigger repaints either so it shouldn't impact performance